mobile/src/main/java/com/truckmuncher/app/vendor/VendorHomeActivity.java

```diff
@@ -128,13 +137,33 @@ public void onServingModeChanged(final boolean enabled, Location currentLocation
         serviceHelper.changeServingState(this, selectedTruck.id, enabled, currentLocation);
 
         final VendorMapFragment fragment = (VendorMapFragment) getSupportFragmentManager().findFragmentById(R.id.vendor_map_fragment);
-
         runOnUiThread(new Runnable() {
             @Override
             public void run() {
                 fragment.setMapControlsEnabled(!enabled);
             }
         });
+
+        // If serving mode is being enabled and they have the item unavailable warning enabled,
+        // we need to check if there are any items marked as unavailable
+        if (enabled && sharedPreferences.getBoolean(getString(R.string.setting_item_unavailable_warning), true)) {
+            // Get menu items that are marked as out of stock for the current truck
+            WhereClause whereClause = new WhereClause.Builder()
+                    .where(PublicContract.Menu.TRUCK_ID, EQUALS, selectedTruck.id)
+                    .and()
+                    .where(PublicContract.Menu.IS_AVAILABLE, EQUALS, 0)
```

Comment:
As an aside, you can pass a boolean here. Also in content values. They get converted correctly. No need to change it though